src/agent/src/config.rs

```diff
@@ -76,6 +77,7 @@ pub struct AgentConfig {
     pub tracing: bool,
     pub endpoints: AgentEndpoints,
     pub supports_seccomp: bool,
+    pub container_policy_path: String,
```

Comment:
Why not `Option<String>` as used in `AgentConfigBuilder`?

Reply:
I originally had the path set to `Option<String>` which lead to me creating a new `get_option_string_value` method, but then I got stuck with the `config_override` macro:
```
error[E0308]: mismatched types
   --> src/config.rs:102:30
    |
102 |             $config.$field = v;
    |                              ^
    |                              |
    |                              expected enum `std::option::Option`, found struct `std::string::String`
    |                              help: try using a variant of the expected enum: `config::_::_serde::__private::Some(v)`
...
187 |         config_override!(agent_config_builder, agent_config, container_policy_path);
    |         ---------------------------------------------------------------------------- in this macro invocation
    |
    = note: expected enum `std::option::Option<std::string::String>`
             found struct `std::string::String`
    = note: this error originates in the macro `config_override` (in Nightly builds, run with -Z macro-backtrace for more info)
```
which I didn't know how to resolve, so I decided the path of least resistance was to use `String` like other fields. If you have recommendations of how to fix the issues I ran into then I'm happy to have another go. Apologies, but my rust is pretty limited.